src/main/java/duke/Task.java

```diff
@@ -0,0 +1,63 @@
+package duke;
+
+/**
+ * Represents a task
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * A constructor for a task
+     *
+     * @param description the task details
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Check if the task has been completed and
+     * return the status icon of the task.
+     *
+     * @return the status icon of the task
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```

Comment:
Instead of using magic strings like "X" and " ", one could consider declaring them as class level constants with meaningful names or use enums